Automattic/wp-calypso
# Trackers Bucketing (based on `sensitive_pixel_options` cookie)

- Implement bucketing (based on a8c-analytics.js)
- Add support for `region` cookie

#### Proposed Changes

* Implement recognition of trackers bucketing cookie (`essential`, `advertising`, `analytics`)
* Implement code that parses `sensitive_pixel_options` to determine if the specific bucket can be tracked
* Replace simple enabled/disabled constant checks for ad trackers with `tracker-buckets.ts` check functions (i.e. `mayWeTrackByTracker`)

#### Testing Instructions

* Ensure `ad-tracking` is enabled on your development environment
* Launch Calypso / Jetpack Cloud
* If not based in CCPA/GDPR region, set a cookie e.g. `document.cookie = 'country_code=NL'` (or any other European country)
* Make sure you have you don't have "sensitive_pixel_option" (legacy) cookie in your session: `document.cookie='sensitive_pixel_option=; expires=Thu, 01 Jan 1970 00:00:01 GMT;'`
* Manually create a bucketing cookie (all buckets accepted): `document.cookie = 'sensitive_pixel_options=%7B%22version%22%3A20201224%2C%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Atrue%2C%22advertising%22%3Atrue%7D%7D';`
* Notice that GA is being loaded and works as usual i.e. if `https://www.googletagmanager.com/gtag/js?id=` scripts are included in page DOM)
* Manually change a bucketing cookie to exclude `analytics` bucket: `document.cookie = 'sensitive_pixel_options=%7B%22version%22%3A20201224%2C%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Afalse%2C%22advertising%22%3Atrue%7D%7D';`
* Notice that GA stopped being loaded to the session, and the rest of trackers (e.g. FB Pixel) is still working i.e. if `https://connect.facebook.net/signals/config/` script is included in page DOM
* Manually change a bucketing cookie to exclude `advertising` bucket: `document.cookie = 'sensitive_pixel_options=%7B%22version%22%3A20201224%2C%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Atrue%2C%22advertising%22%3Afalse%7D%7D';`
* Notice that GA works again, but the rest of trackers (i.e. Facebook Pixel) does not load anymore


To Do:
- [x] Decide if Jetpack GA does use remarketing or not (p1666809890242429-slack-C01264051NE) and update the tracker bucket accordingly


Related: pbNhbs-3sm-p2
